daemon/cmd/kube_proxy_replacement.go

```diff
@@ -69,10 +69,6 @@ func initKubeProxyReplacementOptions(sysctl sysctl.Sysctl, tunnelConfig tunnel.C
 	}
 
 	if option.Config.EnableNodePort {
-		if option.Config.EnableIPSec {
-			return fmt.Errorf("IPSec cannot be used with BPF NodePort")
-		}
-
```

Comment:
From a look at what additional feature combinations this allows (with features that depend on `EnableNodePort`, and thus were previously incompatible), the two that stand out are:
* XDP NodePort acceleration
* BPF Masquerade (and all of its dependencies, like Egress Gateway)

Should we explicitly reject these two, until we have fully understood the implications and added sufficient test coverage?

Reply:
Yeah, it's probably best. Should I already disable for WireGuard or do we have coverage in that case?

Reply:
I'm note quite parsing
>Should I already disable for WireGuard or do we have coverage in that case?

You mean whether WireGuard should also conflict against XDP and/or BPF Masquerade? I think that ship has sailed already, breaking potentially existing configs is a much tougher decision. Coverage-wise, I see that we have WireGuard + EGW in the e2e test.

I would also expect that those features have much more user exposure to WireGuard already (because IPsec was not an option until now).

Reply:
`--enable-host-port` seems like another candidate that's causing problems: https://github.com/cilium/cilium/issues/32897#issuecomment-2157263338